### PR TITLE
Eth: example eth to tft to deploy vm

### PR DESCRIPTION
**Pull request by brandonpille**
Completes #63:
- Added example
- Added documentation

### CODE REVIEW

File: server/pkg/tfchain/client.go [hunk 1]
```diff
@@ -642,3 +655,34 @@ func (c *Client) SwapToStellar(ctx context.Context, conState jsonrpc.State, args
...
+	state := State(conState)
+	if state.client == nil {
+		return pkg.ErrClientNotConnected{}
+	}
+	_, err := tfchainBridgeAddressFromNetwork(state.network)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < int(300); i++ {
```
Comment (DylanVerstraete) [inline server/pkg/tfchain/client.go:669]: make this a `const`

File: server/pkg/tfchain/client.go [hunk 2]
```diff
@@ -642,3 +655,34 @@ func (c *Client) SwapToStellar(ctx context.Context, conState jsonrpc.State, args
 
 	return state.client.SwapToStellar(state.identity, args.TargetStellarAddress, *args.Amount)
 }
+
+func (c *Client) AwaitTransactionOnTfchainBridge(ctx context.Context, conState jsonrpc.State, memo string) error {
```
Comment (DylanVerstraete) [inline server/pkg/tfchain/client.go:659]: `ctx` is an unused parameter, can we also include that if the context is canceled, this function exits?

**Review by DylanVerstraete — CHANGES_REQUESTED** [review]

**Comment by robvanmieghem** [comment]
> While the code looks great, I think the example should not ask for a Stellar and Tfchain key.
> 
> The main reason we want to have this example is to have a seamless experience for an ethereum user having only an ethereum key. All the other keys should be generated on the fly as they are not provided by the user.

Agree but this PR has some necessary changes and value without that functionality too. I would merge this PR and leave that functionality as a TODO in the github issue.

**Comment by robvanmieghem** [comment]
Since the main purpose of this example are demo's, I would default all networks to production.
- The Ethereum Goerli network is marked as deprecated, it requires special knowledge how to configure your Ethereum client and getting Goerli Eth.
- Stellar testnet is reset regularly 
- Testnet bridges are not guaranteed to be up
- devnet tfchain and grid are subject to frequent changes which might cause downtime and incompatibilities  and we don't want to make devnet a 'production' version  which it will become if it is used for demo's.

In other words, testnets and devnets are for developers

This will also make the manual simpler

File: server/pkg/tfchain/client.go (hunk 1 above)
Comment (brandonpille) [reply server/pkg/tfchain/client.go:669]: > While the code looks great, I think the example should not ask for a Stellar and Tfchain key.
> 
> The main reason we want to have this example is to have a seamless experience for an ethereum user having only an ethereum key. All the other keys should be generated on the fly as they are not provided by the user.

It makes sense to create the account for Stellar but not really the account on Tfchain. Because the user will have to cancel the deployment later, right? Or he might want to get information on the deployment later and for that he needs the tfchain account.

Comment (DylanVerstraete) [reply server/pkg/tfchain/client.go:669]: The user will need to extract the key (some method needs to exist to export a tfchain private key). The idea is that the user only has an ethereum key, if we introduce the concept that he also needs to tfchain key we will make it harder to complete this flow.

File: server/clients/stellar/transfer.go [hunk 3]
```diff
@@ -146,43 +147,48 @@ func (c *Client) GetTfchainBridgeAddress() (string, error) {
...
+				decodedMemo, err := base64.StdEncoding.DecodeString(tx.Memo)
+				if err == nil {
+					hexDecodedMemo := hex.EncodeToString(decodedMemo)
+					if hexDecodedMemo == memo {
+						return nil
+					}
 				}
 			}
+		case <-ctx.Done():
+			return nil
```
Comment (robvanmieghem) [inline server/clients/stellar/transfer.go:174]: return ctx.Err()

This way the calling code will not continue if it checks the error. Normally cancelled contexts are not checked after every call where a context is passed but the error is.

File: examples/eth/deploy_vm_with_eth.v [hunk 4]
```diff
@@ -137,10 +137,10 @@ fn main() {
 	eth_url := fp.string('eth-node', 0, '${goerli_node_url}', 'The url of the ethereum node to connect to.')
```
Comment (robvanmieghem) [inline examples/eth/deploy_vm_with_eth.v:137]: I would also default this to the mainnet Ethereum node ` ws://185.69.167.224:8546`

**Review by robvanmieghem — CHANGES_REQUESTED** [review]

**Review by robvanmieghem — APPROVED** [review]
